src/weather.cpp

```diff
@@ -0,0 +1,224 @@
+// Header
+#include "weather.hpp"
+#include "time.h"
+
+
+// stlib
+#include <vector>
+#include <algorithm>
+
+Texture Weather::weather_screens;
+
+bool Weather::init()
+{
+	// Load shared texture
+	weather_screens.load_from_file(textures_path("WeatherSplash.png"));
+
+	// The position (0,0) corresponds to the center of the texture
+	float wr = weather_screens.width * 0.5;
+	float hr = weather_screens.height * 0.5;
+	float indiv_splash_width = 1000.f;
+
+	vertices[0].position = { -indiv_splash_width / 2, +hr, 0.f };
+	vertices[1].position = { +indiv_splash_width / 2, +hr, 0.f };
+	vertices[2].position = { +indiv_splash_width / 2, -hr, 0.f };
+	vertices[3].position = { -indiv_splash_width / 2, -hr, 0.f };
+
+	glGenBuffers(1, &mesh.vbo);
+	glGenBuffers(1, &mesh.ibo);
+
+	// Vertex Array (Container for Vertex + Index buffer)
+	glGenVertexArrays(1, &mesh.vao);
+	if (gl_has_errors())
+		return false;
+
+	// Loading shaders
+	if (!effect.load_from_file(shader_path("textured.vs.glsl"), shader_path("weather.fs.glsl")))
+		return false;
+
+	// Setting initial values, scale is negative to make it face the opposite way
+	// 1.0 would be as big as the original texture
+	m_scale.x = 1.2;
+	m_scale.y = 1.2;
+	m_position.x = 500;
+	m_position.y = 500;
+	weather_timer = 0;
+	srand(time(NULL));
+	game_month = 0; 
+	curr_weather_loc = 0;
+	return true;
+}
+
+// Releases all graphics resources
+void Weather::destroy()
+{
+	glDeleteBuffers(1, &mesh.vbo);
+	glDeleteBuffers(1, &mesh.ibo);
+	glDeleteBuffers(1, &mesh.vao);
+
+	glDeleteShader(effect.vertex);
+	glDeleteShader(effect.fragment);
+	glDeleteShader(effect.program);
+}
+
+void Weather::update(float ms, int level, CurrentTime game_time) {
+	if (level >= 3) {
+		weather_timer += ms / 1000.f;
+		if (game_month != game_time.month) {
+			game_month = game_time.month;
+			determine_sky();
+		}
```

Comment:
could we make this happen more slowly? it feels like the weather changes really quickly

Reply:
Yeah it's because the month change really fast. Was chatting with Jordan about it too, thinking of having them persist for 6mo instead of 3-4 month each.